src/modules/account/hooks/useAccounts.js

```diff
@@ -22,11 +22,24 @@ export function useAccounts() {
   const getAccountByPublicKey = (pubkey) =>
     accounts.find((account) => account.metadata.pubkey === pubkey);
 
+  const setNonceByAccount = useCallback(
+    (address, nonce, transactionHex) => dispatch(setAccountNonce(address, nonce, transactionHex)),
+    []
+  );
+
+  const getNonceByAccount = (address) => {
+    const accountNonceMap = useSelector(selectAccountNonce)[address] ?? {};
```

Comment:
We should not call hooks inside of regular functions. They must be top layer in a component or custom hook. We might want to install [eslint](https://www.npmjs.com/package/eslint-plugin-react-hooks) for to prevent side effects that can happen from this. But i have not seen this patterned use anywhere else in our code (we are bad at passing in dependencies thought 😓)

Reply:
Probably this is resulting in sometimes storing undefined 🤔 
<img width="546" alt="image" src="https://github.com/LiskHQ/lisk-desktop/assets/8784876/6bcb8a71-8372-40b6-a155-d47b965cc7a3">

Reply:
Fixed